library/Opus/Doi/datacite.xslt

```diff
@@ -109,9 +140,11 @@
             </xsl:if>
 
             <xsl:if test="PersonEditor">
-                <xsl:element name="contributors">
-                    <xsl:apply-templates select="PersonEditor"/>
-                </xsl:element>
+				<xsl:if test="PersonAuthor or @CreatingCorporation">
```

Comment:
Warum wurde an dieser Stelle (Erzeugung des XML-Elements `contributor`) eine zusätzliche ìf`-Prüfung eingebaut? Die Elemente `PersonAuthor` bzw. `@CreatingCorporation` werden bei der Ausgabe von `PersonEditor` gar nicht berücksichtigt.

Reply:
Könnte/sollte man kommentieren. Ich interpretiere es so. Wenn es einen Editor gibt wird dieser als Creator verwendet, wenn es KEINEN Autoren bzw. eine CreatingCorporation gibt. In dem Fall, dass der Editor nicht als Creator verwendet wurde, soll er hier als Contributor eingefügt werden, also wenn Editor existiert und Autor bzw. CreatingCorporation auch. :-)

Reply:
Ich bin über diese Stelle gestolpert, weil man ja beide Zeilen auch problemlos zu einer Zeile zusammenfassen kann. Aber vielleicht gab es eine Absicht für dieses Vorgehen. Daher meine Nachfrage.

Reply:
Inhaltlich geht es darum, dass PersonEditor nur dann als Contributor ausgegeben werden soll, wenn das Element nicht bereits als Creator verwendet wird. Die Ausgabe als Creator erfolgt, wenn weder PersonAuthor noch CreatingCorporation vorhanden sind. 

Die Verwendung von zwei aufeinander folgenden if-Abfragen in Zeile 142 und 143 hat sich aus der Anlehnung an den Code für das "creators"-Element in den Zeilen 64 ff. ergeben. Es spricht nichts dagegen, das in eine Zeile zu packen.

Reply:
Ok, danke für das Feedback. Ich würde dann alles in ein if-Statement zusammenpacken. Dann stolpert man später nicht nochmal drüber.